lib/Item/ValueLoader/TaxonValueLoader.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Layouts\Sylius\Item\ValueLoader;
+
+use Netgen\Layouts\Item\ValueLoaderInterface;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Throwable;
+
+final class TaxonValueLoader implements ValueLoaderInterface
+{
+    /**
+     * @param TaxonRepositoryInterface<TaxonInterface> $taxonRepository
+     */
+    public function __construct(private readonly TaxonRepositoryInterface $taxonRepository) {}
+
+    public function load($id): ?TaxonInterface
+    {
+        try {
+            $taxon = $this->taxonRepository->find($id);
+        } catch (Throwable) {
+            return null;
+        }
+
+        return $taxon instanceof TaxonInterface ? $taxon : null;
```

Comment:
@igorkrz Is this check needed? Taxon repository will either return TaxonInterface or null so you can just return the result.

Reply:
Yes, but PHPStan does not know that 😂